proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1205 @@ If WiFi credentials are automatically generated by the IVI system, the password
+)
 
-* If `MultiplexTransportConfig.setCanAutoJoinWiFiNetwork( true )` and mobile device __DOES NOT__ support WiFi transport APIs provided by platform SDK then `DeviceInfo.wifiConfigurationInfo.canAutoJoinWiFiNetwork = false`
 
-* If `MultiplexTransportConfig.setCanHostWiFiNetwork( false )` and mobile device __DOES__ support WiFi transport APIs provided by platform SDK then `DeviceInfo.wifiConfigurationInfo.canHostWiFiNetwork = false`
+### RPCs Encryption
 
-* If `MultiplexTransportConfig.setCanAutoJoinWiFiNetwork( false )` and mobile device __DOES__ support WiFi transport APIs provided by platform SDK then `DeviceInfo.wifiConfigurationInfo.canAutoJoinWiFiNetwork= false`
+Wifi credentials considered as sensitive data, and should not be shared without encryption.
+[SDL 0207 - RPC message protection](https://github.com/smartdevicelink/sdl_evolution/issues/634) SDL feature.
 
-6. The proxy should use the above mentioned mobile APIs for WiFi if corresponding `DeviceInfo.wifiConfigurationInfo` is set.
+RPC message protection is controlled completely by Policy table.
 
-* If `DeviceInfo.wifiConfigurationInfo.canHostWiFiNetwork = true` and supported secondary transport received in `Start Service Ack` protocol message is `TCP_WIFI_AP_MODE` only then the proxy can use `JoinWiFiNetwork` RPC.
+It is adviced to restrict `JoinNetwork`, to be sent only by encrypted RPC service, but it is up to particular OEM. SDL does not enforce `JoinNetwork`, to be send only by encrypted RPC service.
 
-* If `DeviceInfo.wifiConfigurationInfo.canAutoJoinWiFiNetwork = true` and supported secondary transport received in `Start Service Ack` protocol message is `TCP_WIFI_STA_MODE` only then the proxy can use `GetWiFiNetworkInfo` RPC.
+If `JoinNetwork` or `OnNetworkConfiguration` should not be protected, there are no additional issues.
+If `JoinNetwork` or `OnNetworkConfiguration` allowed only via protected RPC service, following rules applied:
 
-7. If app developers use this API to enable WiFi connection feature, they should provide below permissions/entitlements:
-*  Android App will need to include WiFi permissions.
-*  iOS App will need to add `Hotspot Configuration` Capability.
-8. The proxy should use appropriate WiFi mobile APIs as soon as `OnSystemRequest` notification with `RequestType` as `WIFI_CONNECTION_REQ` is received.
-9. The proxy should rely on error callbacks and return values to know whether WiFi connection using platform APIs was successful or failed. 
+ - Reject `JoinNetwork` from Mobile if it is sent via Unprotected service. (guaranteed by SDL 0207)
+ - Reject `JoinNetwork` from HMI if appropriate mobile has not start protected service yet(result code : REJECT)
+ - SDL should not send `OnNetworkConfiguration` to HMI until appropriate mobile won't start secure RPC service.<br>
+__Note:__ Such rule required because HMI `OnNetworkConfiguration` for HMI can be trigger for sending `JoinNetwork` to mobile.

```

Comment:
```suggestion
__Note:__ Such rule is required because HMI `OnNetworkConfiguration` for HMI can be trigger for sending `JoinNetwork` to mobile.

```